### PR TITLE
NAS-105016 / 12.0 / OpenStack Swift support for cloud sync

### PR DESCRIPTION
**Pull request by JGoutin**
OpenStack Swift support for cloud sync.

### CODE REVIEW

File: src/middlewared/middlewared/rclone/remote/swift.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        Str("application_credential_id", default="",
+            title="Application Credential ID (OS_APPLICATION_CREDENTIAL_ID)"),
+        Str("application_credential_name", default="",
+            title="Application Credential Name "
+                  "(OS_APPLICATION_CREDENTIAL_NAME)"),
+        Str("application_credential_secret", default="",
+            title="Application Credential Secret "
+                  "(OS_APPLICATION_CREDENTIAL_SECRET)"),
+        Str("auth_version", default="",
+            title="AuthVersion - set to (1,2,3) if your auth URL has no "
```
Comment (themylogin) [inline src/middlewared/middlewared/rclone/remote/swift.py]: Have you tested what does `set to (1,2,3)` mean? Set to that exact string (but it is `Type: int`)? Or set to any of these values?

Comment (JGoutin) [reply src/middlewared/middlewared/rclone/remote/swift.py]: This is the API version used, the value can be `1`, `2` or ` (or `0` by default). This is used only on some clouds.
And yes, [it is a Int](https://rclone.org/swift/#swift-auth-version), I'll fix this.

**Comment by themylogin** [comment]
@JGoutin would you have some time to ensure that I've designed corresponding UI form properly?  https://jira.ixsystems.com/browse/NAS-105021

**Comment by JGoutin** [comment]
> @JGoutin would you have some time to ensure that I've designed corresponding UI form properly? https://jira.ixsystems.com/browse/NAS-105021

This looks good for me. :+1:

**Pull request merged**